src/wallet-user/Tokenization.tsx

```diff
@@ -30,20 +30,20 @@ interface FormValues {
 }
 
 export default function Tokenization(props: Props) {
-    const { currentAddress, colorTheme } = useCommonContext();
+    const { addresses, colorTheme } = useCommonContext();
     const { refreshRequests, createTokenRequest } = useUserContext();
     const [state, setState] = useState(props.initialState !== null ? props.initialState : State.START)
     const { control, handleSubmit, formState: { errors }, reset } = useForm<FormValues>();
 
     const submit = useCallback((formValues: FormValues) => {
-        console.log(formValues);
         (async function() {
             const attributes = [
                 `${DEFAULT_LEGAL_OFFICER}`,
                 `${formValues.requestedTokenName}`,
                 `${formValues.bars}`
             ];
             const signedOn = moment();
+            const currentAddress = addresses!.currentAddress!.address;
```

Comment:
Just wondering if there is a way to factor out the expression `addresses?.currentAddress?.address` which appear quite often in the code ?

Reply:
Well, in the beginning there was a `currentAddress` field in the context which was an "alias" for that expression. I removed it because I found it quite error prone (risk of discrepancy between that field and the one in `Addresses`). A possible solution could be to turn `Addresses` into a class (currently an interface) and exposing `currentAddress` as a property.

Reply:
One suggestion: rename `AccountAddress` into `Account`:

```
```

Reply:
Will do.

Reply:
Just created a subtask for doing this later. Otherwise, I may create a "rebase hell" with my current work... :-)